gadget/gadget.go

```diff
@@ -206,6 +208,106 @@ type VolumeUpdate struct {
 	Preserve []string       `yaml:"preserve"`
 }
 
+// MappedVolumeDeviceHeuristics is a set of heuristics to use to try and map a
```

Comment:
heuristics is a bit strange here, because refers originally more to strategies than data, maybe we should use "traits" instead and explain here that they are as clues in heuristic ways